airflow/providers/microsoft/azure/hooks/powerbi.py

```diff
@@ -0,0 +1,323 @@
+                f"Unable to fetch the details of dataset refresh with Request Id: {request_id}"
+            )
+
+        request_id_index = request_ids.index(request_id)
+        refresh_details = refresh_histories[request_id_index]
+
+        return refresh_details
+
+    def wait_for_dataset_refresh_status(
+        self,
+        *,
+        expected_status: str,
+        request_id: str,
+        dataset_id: str,
+        group_id: str,
+        check_interval: int = 60,
+        timeout: int = 60 * 60 * 24 * 7,
+    ) -> bool:
+        """
+        Wait until the dataset refresh of given request ID has reached the expected status.
+
+        :param expected_status: The desired status to check against a dataset refresh's current status.
+        :param request_id: Request id for the dataset refresh request.
+        :param check_interval: Time in seconds to check on a dataset refresh's status.
+        :param timeout: Time in seconds to wait for a dataset to reach a terminal status or the expected status.
+        :return: Boolean indicating if the dataset refresh has reached the ``expected_status`` before the timeout.
+        """
+        dataset_refresh_details = self.get_refresh_details_by_request_id(
+            dataset_id=dataset_id, group_id=group_id, request_id=request_id
+        )
+        dataset_refresh_status = dataset_refresh_details.get(PowerBIDatasetRefreshFields.STATUS.value)
+
+        start_time = time.monotonic()
+
+        while (
+            dataset_refresh_status not in PowerBIDatasetRefreshStatus.TERMINAL_STATUSES
+            and dataset_refresh_status != expected_status
+        ):
+            # Check if the dataset-refresh duration has exceeded the ``timeout`` configured.
+            if start_time + timeout < time.monotonic():
+                raise PowerBIDatasetRefreshException(
+                    f"Dataset refresh has not reached a terminal status after {timeout} seconds"
+                )
+
+            time.sleep(check_interval)
+
+            dataset_refresh_details = self.get_refresh_details_by_request_id(
+                dataset_id=dataset_id, group_id=group_id, request_id=request_id
+            )
+            dataset_refresh_status = dataset_refresh_details.get(PowerBIDatasetRefreshFields.STATUS.value)
+
+        return dataset_refresh_status == expected_status
+
+    def trigger_dataset_refresh(self, *, dataset_id: str, group_id: str) -> str:
+        """
+        Triggers the Power BI dataset refresh.
+
+        :param dataset_id: The dataset ID.
+        :param group_id: The workspace ID.
+
+        :return: Request ID of the dataset refresh request.
+        """
+        # Start dataset refresh
+        self.log.info("Starting dataset refresh.")
+        request_id = self.refresh_dataset(dataset_id=dataset_id, group_id=group_id)
+
+        return request_id
+
+    def _send_request(self, request_type: str, url: str, **kwargs) -> requests.Response:
```

Comment:
This code is not needed as this is already implemented by the HttpxRequestAdapter of the Microsoft Kiota library, please check the source code of the KiotaRequestAdapterHook.  Also the KiotaRequestAdapterHook implements a method call run with parameter, which respect a bit the convention agreed in Airflow for Hooks and which actually allows you to do what you've implemented here, except it just delegates that logic to the Microsoft Kiota library.